core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/GroupIterator.java

```diff
@@ -485,11 +480,19 @@ private Supplier<Predicate<Value>> createDistinctValueTest(AggregateOperator ope
 		return operator.isDistinct() ? DistinctValues::new : ALWAYS_TRUE_VALUE_SUPPLIER;
 	}
 
-	private QueryStepEvaluator precompileArg(AggregateOperator operator) {
+	private QueryStepEvaluator precompileUnaryArg(AggregateOperator operator) {
 		ValueExpr ve = ((UnaryValueOperator) operator).getArg();
 		return new QueryStepEvaluator(strategy.precompile(ve, context));
 	}
 
+	private NAryQueryStepEvaluator precompileNAryArg(AggregateOperator operator) {
+		List<ValueExpr> args = ((NAryValueOperator) operator).getArguments();
+		return new NAryQueryStepEvaluator((index) -> {
+			ValueExpr ve = args.get(index);
+			return strategy.precompile(ve, context);
```

Comment:
I believe this ends up calling precompile more than just once per argument. 

Maybe something like this would work:

```
	private NAryQueryStepEvaluator precompileNAryArg(AggregateOperator operator) {
		List<ValueExpr> args = ((NAryValueOperator) operator).getArguments();
		List<QueryValueEvaluationStep> precompiledArgs = new ArrayList<>(args.size());
		for (ValueExpr arg : args) {
			precompiledArgs.add(strategy.precompile(arg, context));
		}
		return new NAryQueryStepEvaluator(precompiledArgs::get);
	}
```